sdk/storage/azure-storage-file-datalake/azure/storage/filedatalake/aio/_file_system_client_async.py

```diff
@@ -720,72 +720,67 @@ async def delete_file(self, file,  # type: Union[FileProperties, str]
+    @distributed_trace_async
+    async def delete_files(
+        self,
+        *files: str,
+        **kwargs) -> List[Optional[HttpResponseError]]:
+        """Marks the specified files or empty directories for deletion.
+
+        The files/empty directories are later deleted during garbage collection.
+
+        If a delete retention policy is enabled for the service, then this operation soft deletes the
+        files/empty directories and retains the files or snapshots for specified number of days.
+        After specified number of days, files' data is removed from the service during garbage collection.
+        Soft deleted files/empty directories are accessible through :func:`list_deleted_paths()`.
+
+        :param str files:
+            The files/empty directories to delete. This can be a single file/empty directory, or multiple values can
+            be supplied, where each value is the name of the file/directory (str).
+
+        :keyword ~datetime.datetime if_modified_since:
+            A DateTime value. Azure expects the date value passed in to be UTC.
+            If timezone is included, any non-UTC datetimes will be converted to UTC.
+            If a date is passed in without timezone info, it is assumed to be UTC.
+            Specify this header to perform the operation only
+            if the resource has been modified since the specified time.
+        :keyword ~datetime.datetime if_unmodified_since:
+            A DateTime value. Azure expects the date value passed in to be UTC.
+            If timezone is included, any non-UTC datetimes will be converted to UTC.
+            If a date is passed in without timezone info, it is assumed to be UTC.
+            Specify this header to perform the operation only if
+            the resource has not been modified since the specified date/time.
+        :keyword bool raise_on_any_failure:
+            This is a boolean param which defaults to True. When this is set, an exception
+            is raised even if there is a single operation failure.
+        :keyword int timeout:
+            The timeout parameter is expressed in seconds.
+        :return: A list containing None for successful operations and HttpResponseError objects for unsuccessful operations.
+        :rtype: List[Optional[HttpResponseError]]
+
+        .. admonition:: Example:
+
+            .. literalinclude:: ../samples/datalake_samples_file_system_async.py
+                :start-after: [START batch_delete_files_or_empty_directories]
+                :end-before: [END batch_delete_files_or_empty_directories]
+                :language: python
+                :dedent: 4
+                :caption: Deleting multiple files or empty directories.
+        """
+        response = await self._container_client.delete_blobs(*files, **kwargs)
+
+        results = []
+        async for item in response:
+            results.append(item)
+
+        errors = []
+        for result in results:
```

Comment:
Sorry if I missed this before but I think everything can be done in one loop? You should be able to build the error list in the `async for` loop I would think.